lib/task-queue/WorkerTask.js

```diff
@@ -0,0 +1,192 @@
+   *                    Returns a promise that resolves when the task is saved
+   */
+  save() {
+    return WorkerTask.save(this);
+  }
+
+  /**
+   * Used by JSON.stringify to serialize the Task object
+   *
+   * @return {Object}
+   *                  Returns the serializable task
+   */
+  toJSON() {
+    return {
+      id: this.id,
+      pageUrl: this.pageUrl,
+      createdAt: this.createdAt,
+      jobStartedAt: this.jobStartedAt,
+      status: this._status,
+      type: this.type,
+    };
+  }
+
+  /**
+   * Creates a new task from an Object
+   *
+   * @param {Object} obj
+   *                 The object containing the task attributes values
+   */
+  static fromObject(obj) {
+    let task = new WorkerTask(obj.pageUrl, obj.type, obj.createdAt);
+    Object.assign(task, {
+      id: obj.id,
+      jobStartedAt: obj.jobStartedAt,
+      _status: obj.status,
+    });
+    return task;
+  }
+
+  /**
+   * A static method to save a task
+   *
+   * @param {Task}      task
+   *                    The task to save
+   * @returns {Promise}
+   *                    Returns a promise that resolves when the task is saved
+   */
+  static save(task) {
+    // SQL to insert a new task
+    const insertSQL = `INSERT INTO moz_tasks (pageUrl, createdAt, jobStartedAt, status, type)
+                       VALUES (:pageUrl, :createdAt, :jobStartedAt, :status, :type);SELECT last_insert_rowid() FROM moz_tasks;`;
+    // SQL to update an existing task
+    const updateSQL = `UPDATE moz_tasks SET pageUrl = :pageUrl, createdAt = :createdAt,
+                       jobStartedAt = :jobStartedAt, status = :status, type = :type WHERE id = :id`;
+    // Query parameters
+    let params = {
+      pageUrl: task.pageUrl,
+      createdAt: task.createdAt,
+      jobStartedAt: task.jobStartedAt,
+      status: task.status,
+      type: task.type
+    };
+
+    let sql = insertSQL;
+
+    // If the task id is set, it's an existing task, so we want to update it.
+    if (task.id) {
+      sql = updateSQL;
+      params.id = task.id;
+    }
+
+    // Spawns a task to execute the SQL query and returns a promise.
+    return Task.spawn(function*() {
+      let storage = new Storage();
+      yield storage.executeCached("Saving task", sql, {params});
+
+      // If the task is new and doesn't have an id we want to
+      // recover the auto-incremented id value from the database.
+      if (!task.id) {
+        yield storage.executeCached("Recovering task id", "select last_insert_rowid() FROM moz_tasks", {callback: (row)=> {
+          task.id = row.getResultByName("last_insert_rowid()");
+        }});
+      }
+      yield storage.closeConnection();
+    });
+  }
+
+  /**
+   * Get a task from the database by it's id
+   *
+   * @param {String}   id
+   *                   The id of the task we want to get
+   * @return {Promise}
+   *                   Returns a promise that resolves to the found task, or null otherwise.
+   */
+  static getById(id) {
```

Comment:
Please make this a `Task.async` function and prefix the function name with `async`